codemods/buttons.js

```diff
@@ -0,0 +1,53 @@
+const visit = require('unist-util-visit');
+const {
+  isMdxBlockElement,
+  removeAttribute,
+  addAttribute,
+} = require('./utils/mdxast');
+
+const VARIANTS = ['btn-primary'];
+
+const buttons = () => (tree) => {
+  visit(
+    tree,
+    (node) => {
+      return (
+        isMdxBlockElement('a', node) &&
+        node.attributes.some(
+          (attr) => attr.name === 'className' && attr.value.includes('btn')
+        )
+      );
+    },
+    (a) => {
+      const { value: className } = a.attributes.find(
+        (attr) => attr.name === 'className'
+      );
+
+      const { value: href } = a.attributes.find((attr) => attr.name === 'href');
+
+      const variant = className
+        .split(/\s+/)
+        .find(
+          (className) =>
+            className.startsWith('btn-') && VARIANTS.includes(className)
+        )
+        .replace('btn-', '');
+
+      if (href) {
+        a.attributes.push({
+          name: 'as',
+          value: 'a',
```

Comment:
By setting this value to an `a`, it means that clicking on the button will do a whole page refresh. We want to try and use Gatsby's `Link` element where possible to provide the best experience our users.

Since that value is more than just a string, the `value` type becomes a bit more complex. After some digging, it looks like this type should handle a JSX _expression_ instead of a plain string:

```js
a.attributes.push({
  type: 'mdxAttribute',
  name: 'as',
  value: {
    type: 'mdxValueExpression',
    value: 'Link',
  }
})
```

This should transform the button into the following JSX:

```js
<Button as={Link} />
```

With this change, we should likely switch the `href` to a `to` prop to be compatible with the `Link` component.

Reply:
this is the information I was looking for! Question: where were you able to learn about this? I've been pouring over the MDX docs and they're pretty sparse.

Reply:
Lets chat in person!